### PR TITLE
moved logic from __init__.py to Predictor class

### PR DESCRIPTION
**Pull request by iskandr**

### CODE REVIEW

File: utopia/predictor.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        Parameters
+        ----------
+        mhc_model_type : type
+            Type of MHC Class I binding prediction model whose
+            initializer must take a list of HLA allele names called 'alleles'
+            and a list of integers called 'epitope_lengths'
+
+        epitope_lengths : list of int
+
+        padding_around_mutation : int, optional
```
Comment (tavinathanson) [inline utopia/predictor.py]: What if you want to include the whole thing? I guess that can be in another PR?

**Comment by tavinathanson** [comment]
Update the single test to work with this?

File: utopia/predictor.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        require_integer(padding_around_mutation, "padding around mutation")
+        self.padding_around_mutation = padding_around_mutation
+
+    def predict(
+            self,
+            variant_collection,
+            hla_alleles,
+            raise_on_variant_effect_error):
+        mhc_model = self.mhc_model_type(
```
Comment (tavinathanson) [inline utopia/predictor.py]: So what exactly is `mhc_model`? I guess it'll be more clear when I see the test code running.

File: utopia/predictor.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        self.mhc_model_type = mhc_model_type
+        self.epitope_lengths = epitope_lengths
+        if min(epitope_lengths) < 1:
+            raise ValueError(
+                "Epitope lengths must be positive integers, got: %s" % (
+                    min(epitope_lengths),))
+        if padding_around_mutation is None:
+            padding_around_mutation = max(epitope_lengths) - 1
+
+        require_integer(padding_around_mutation, "padding around mutation")
```
Comment (tavinathanson) [inline utopia/predictor.py]: I thought you were moving away from `require` statements? This one feels funky because it's so deep in the method. (If we had a require, I think I'd prefer `require_integer_or_none` or `require_integer(none_okay=True` or something?

**Comment by tavinathanson** [comment]
Looks good to me!

File: utopia/predictor.py (hunk 2 above)
Comment (iskandr) [inline utopia/predictor.py]: It's the binding predictor.

File: utopia/predictor.py (hunk 3 above)
Comment (iskandr) [inline utopia/predictor.py]: I didn't mean to make a blanket rule about "always have requires" or "never have requires", just that I only want to put them in places where the user might pass in the wrong type.
